src/lib/operators/join_sort_merge.cpp

```diff
@@ -182,7 +189,8 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractJoinOperatorImpl {
     // Note: this is only provisional. There should be a reasonable calculation here based on hardware stats.
     size_t chunk_count_left = _sort_merge_join.input_table_left()->chunk_count();
     size_t chunk_count_right = _sort_merge_join.input_table_right()->chunk_count();
-    return static_cast<size_t>(std::pow(2, std::floor(std::log2(std::max(chunk_count_left, chunk_count_right)))));
```

Comment:
Reason for that change? Since the chunk counts can never be negative, there shouldn't be values smaller than 2^0. One problem that might occur are empty tables with chunk counts of 0 such that `log2()` fails, but that's still not covered (can that even happen?).

Reply:
I think I changed this precisely because Tables might be empty, i.e., containing no Chunks. Yes, imo (and I *think* @mrks agrees) this is a very valid state and operators need to handle it properly

Reply:
Not what I meant. I see the point for empty chunks, no question.

But what the current change does it to ensure that at least a value of 1 is returned. However, 2^0 is 1, so that's ensured anyhow. Of course assuming we will never see negative chunk counts (should be save to assume). Another problem that is currently not handled (and hasn't been before) is the `log(0)` case, which would fail (both tables have 0 chunks). So what I am proposing is to move the` max(1)` to the chunk count max: `std::log2(std::max({1, chunk_count_left, chunk_count_right})`. For all values >= 1, the following `2^n` will always yield a valid chunk count.

Reply:
gottcha. changed.